File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.cpp
class Solution {
public:
    int countCompleteSubarrays(vector<int>& nums) {
        unordered_set<int>st;
        for(auto i: nums) st.insert(i);

        int sizeOfDinstinctElement=st.size();
        // st.clear();
        int n=nums.size();
        int count=0;


        for(int i=0;i<=n-sizeOfDinstinctElement;i++){
            unordered_set<int>st2;
            for(int j=i;j<n;j++){
                st2.insert(nums[j]);
                if(st2.size()==sizeOfDinstinctElement) count++;
                if(st2.size()>sizeOfDinstinctElement) break;
            }
             
        }
        return count;
    }
};

File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.java
class Solution {
    public int countCompleteSubarrays(int[] nums) {
        int n=nums.length;

        Set<Integer>st=new HashSet<>();
        for(int i=0;i<n;i++) st.add(nums[i]);
        int distintSize=st.size();
        st.clear();

        int cnt=0;
        for(int i=0;i<=n-distintSize;i++){
            for(int j=i;j<n;j++){
                st.add(nums[j]);
                if(st.size()==distintSize) cnt++;
                if(st.size()>distintSize) break;
            }
            st.clear();
        }
        return cnt;
    }
}